### PR TITLE
feat: Anti-raid-nuke report generator

### PR DESCRIPTION
**Pull request by JPBM135**
Revive the markdown report system, now localised and improved!!

### CODE REVIEW

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts [hunk 1]
```diff
@@ -83,6 +105,7 @@ export async function handleAntiRaidNuke(
 				lng: locale,
 			})}\n\n${prefixedParameterStrings.join('\n')}`,
 		});
+		return;
 	}
```
Comment (iCrawl) [inline packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:109]: Can't mark beyond this point, but can't we just throw here instead?

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts [hunk 2]
```diff
@@ -143,14 +166,19 @@ export async function handleAntiRaidNuke(
...
+			content: i18next.t(`command.mod.anti_raid_nuke.common.${collectedInteraction ? 'cancel' : 'timeout'}`, {
 				lng: locale,
 			}),
 			components: [],
-		});
-	} else if (collectedInteraction?.customId === banKey || collectedInteraction?.customId === dryRunKey) {
+		};
+		if (!collectedInteraction) {
+			await interaction.editReply(payload);
+			return;
```
Comment (iCrawl) [inline packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:178]: Same as above.

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts [hunk 3]
```diff
@@ -8,12 +9,17 @@ import {
 	type GuildMember,
 	type ModalSubmitInteraction,
 	type Snowflake,
+	ActionRowData,
+	ButtonComponent,
+	Message,
```
Comment (iCrawl) [inline packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:14]:
```suggestion
	type ActionRowData,
	type ButtonComponent,
	type Message,
```

File: packages/yuudachi/locales/en-US/translation.json [hunk 4]
```diff
@@ -459,5 +456,89 @@
 				"description": "• Moderator: {{- moderator}}\n• Hit count: {{count}}"
 			}
 		}
+	},
+	"formatters": {
+		"formatting": {
+			"guildmember": "{{-id}}| Joined: {{-join}} | Created: {{-creation}} | {{-tag}}"
```
Comment (iCrawl) [inline packages/yuudachi/locales/en-US/translation.json:462]:
```suggestion
			"guild_member": "{{- id}} | Joined: {{- join}} | Created: {{- creation}} | {{- tag}}"
```

Also make sure to update all instances that used `guildmember` to `guild_member`

File: packages/yuudachi/locales/en-US/translation.json [hunk 5]
```diff
@@ -459,5 +456,89 @@
...
+				"banned_zero": "No members were banned",
+				"failed": "`{{count}}` members failed to be banned",
+				"failed_zero": "All members were banned",
+				"ratio": "Ratio: `{{ratio}}`"
+			},
+			"parameters": {
+				"title": "**Parameters**",
+				"avatar": "**Matching avatar:** `{{avatar}}`",
+				"pattern": {
+					"title": "**Pattern:** `{{-pattern}}`",
```
Comment (iCrawl) [inline packages/yuudachi/locales/en-US/translation.json:489]:
```suggestion
					"title": "**Pattern:** `{{- pattern}}`",
```

File: packages/yuudachi/locales/en-US/translation.json [hunk 6]
```diff
@@ -459,5 +456,89 @@
...
+					"confusables": "Pattern is resistant to confusables",
+					"insensitive": "Pattern is case-insensitive",
+					"full_match": "Pattern is restricted to full matches"
+				},
+				"created": {
+					"title": "**Creation range:**",
+					"after": "Created after: `{{-after}}`",
+					"before": "Created before: `{{-before}}`"
```
Comment (iCrawl) [inline packages/yuudachi/locales/en-US/translation.json:497]:
```suggestion
					"after": "Created after: `{{- after}}`",
					"before": "Created before: `{{- before}}`"
```

File: packages/yuudachi/locales/en-US/translation.json [hunk 7]
```diff
@@ -459,5 +456,89 @@
...
+				},
+				"joined": {
+					"title": "**Join range:**",
+					"after": "Joined after: `{{-after}}`",
+					"before": "Joined before: `{{-before}}`"
```
Comment (iCrawl) [inline packages/yuudachi/locales/en-US/translation.json:502]:
```suggestion
					"after": "Joined after: `{{- after}}`",
					"before": "Joined before: `{{- before}}`"
```

File: packages/yuudachi/locales/en-US/translation.json [hunk 8]
```diff
@@ -459,5 +456,89 @@
...
+					"2": "Filtering members only",
+					"3": "Filtering members and pattern resistance"
+				},
+				"file": {
+					"title": "**File:** `{{file}}`",
+					"placeholder": "<CDN link>"
+				},
+				"days_zero": "Not deleting any messages",
+				"days_one": "Deleting messages sent in the last day",
+				"days_other": "Deleting messages sent in the last {{count}} days",
```
Comment (iCrawl) [inline packages/yuudachi/locales/en-US/translation.json:517]:
```suggestion
				"days_other": "Deleting messages sent in the last `{{count}}` days",
```

File: packages/yuudachi/locales/en-US/translation.json [hunk 9]
```diff
@@ -459,5 +456,89 @@
...
+				"title": "**Cases**",
+				"range": "From case `#{{from}}` to case `#{{to}}`",
+				"none": "No case created for this Nuke",
+				"log_message": "**Log message:** [Beam me up, Yuki]({{- message}})",
+				"log_message_none": "**Log message:** `No log message created`"
+			},
+			"successes": {
+				"title": "**Successes**",
+				"none": "No successes, something went wrong",
+				"tables": ["Case Id", "Member Id", "Username"]
```
Comment (iCrawl) [inline packages/yuudachi/locales/en-US/translation.json:531]:
```suggestion
				"tables": ["Case id", "Member id", "Username"]
```

File: packages/yuudachi/locales/en-US/translation.json [hunk 10]
```diff
@@ -459,5 +456,89 @@
...
+			},
+			"failures": {
+				"title": "**Failures**",
+				"none": "No failures, everything went fine",
+				"tables": ["Member Id", "Username", "Error"]
```
Comment (iCrawl) [inline packages/yuudachi/locales/en-US/translation.json:536]:
```suggestion
				"tables": ["Member id", "Username", "Error"]
```

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts [hunk 11]
```diff
@@ -1,3 +1,4 @@
+import { performance } from 'perf_hooks';
```
Comment (iCrawl) [inline packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:1]:
```suggestion
import { performance } from 'node:perf_hooks';
```

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts [hunk 12]
```diff
@@ -59,15 +73,23 @@ export async function validateMemberIds(
 	};
 }
 
+function parseDate(date: string | undefined) {
+	if (!date) {
+		return undefined;
+	}
+
+	return dayjs(resolveTimestamp(date)).format(DATE_FORMAT_WITH_SECONDS);
+}
```
Comment (iCrawl) [inline packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:82]:
```suggestion
function parseDate(date?: string | null | undefined) {
	return date ? dayjs(resolveTimestamp(date)).format(DATE_FORMAT_WITH_SECONDS) : null;
}
```

File: packages/yuudachi/src/functions/formatters/generateAntiRaidNukeReport.ts [hunk 13]
```diff
@@ -0,0 +1,251 @@
+import { ms } from '@naval-base/ms';
+import dayjs from 'dayjs';
+import { codeBlock, Guild, hyperlink, User } from 'discord.js';
```
Comment (iCrawl) [inline packages/yuudachi/src/functions/formatters/generateAntiRaidNukeReport.ts:3]:
```suggestion
import { codeBlock, type Guild, hyperlink, type User } from 'discord.js';
```

File: packages/yuudachi/src/functions/formatters/generateAntiRaidNukeReport.ts [hunk 14]
```diff
@@ -0,0 +1,251 @@
...
+import { blockquote, checkbox, emptyLine, heading, horizontalRule, list, table } from './markdownUtils.js';
+import type { ArgsParam } from '../../Command.js';
+import { DATE_FORMAT_WITH_SECONDS } from '../../Constants.js';
+import type { AntiRaidNukeMode } from '../../commands/moderation/sub/anti-raid-nuke/coreCommand.js';
+import type { AntiRaidNukeCommand } from '../../interactions/index.js';
+import { kSQL } from '../../tokens.js';
+import type { AntiRaidNukeResult } from '../anti-raid/blastOff.js';
+import { formatAntiRaidResultsToAttachment } from '../logging/formatMembersToAttachment.js';
+
+export type AntiRaidArgsUnion = ArgsParam<typeof AntiRaidNukeCommand>['file'] &
```
Comment (iCrawl) [inline packages/yuudachi/src/functions/formatters/generateAntiRaidNukeReport.ts:16]:
```suggestion
export type AntiRaidNukeArgsUnion = ArgsParam<typeof AntiRaidNukeCommand>['file'] &
```

File: packages/yuudachi/src/functions/formatters/generateAntiRaidNukeReport.ts [hunk 15]
```diff
@@ -0,0 +1,251 @@
...
+	ArgsParam<typeof AntiRaidNukeCommand>['filter'] &
+	ArgsParam<typeof AntiRaidNukeCommand>['modal'];
+
+export type FormatterArgs = Partial<AntiRaidArgsUnion> & {
```
Comment (iCrawl) [inline packages/yuudachi/src/functions/formatters/generateAntiRaidNukeReport.ts:20]:
```suggestion
export type FormatterArgs = Partial<AntiRaidNukeArgsUnion> & {
```

File: packages/yuudachi/src/functions/formatters/generateAntiRaidNukeReport.ts [hunk 16]
```diff
@@ -0,0 +1,251 @@
...
+
+export async function generateAntiRaidNukeReport(
+	guild: Guild,
+	user: User,
+	result: AntiRaidNukeResult[],
+	args: FormatterArgs,
+	locale: string,
+) {
+	const parts = [];
+	const t = i18next.t;
```
Comment (iCrawl) [inline packages/yuudachi/src/functions/formatters/generateAntiRaidNukeReport.ts:70]: Rather not do that 👀

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts [hunk 17]
```diff
@@ -59,15 +73,23 @@ export async function validateMemberIds(
...
+
 export async function handleAntiRaidNuke(
 	interaction: ChatInputCommandInteraction<'cached'> | ModalSubmitInteraction<'cached'>,
 	members: Collection<Snowflake, GuildMember>,
-	locale: string,
+	mode: AntiRaidNukeMode,
 	parameterStrings: string[],
-	reason?: string,
-	days?: number,
+	args: Partial<AntiRaidArgsUnion>,
```
Comment (iCrawl) [inline packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:89]:
```suggestion
	args: Partial<AntiRaidNukeArgsUnion>,
```

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts [hunk 18]
```diff
@@ -8,12 +9,17 @@ import {
...
+	Message,
 } from 'discord.js';
 import i18next from 'i18next';
 import { nanoid } from 'nanoid';
-import { DATE_FORMAT_LOGFILE } from '../../../../Constants.js';
+import { DATE_FORMAT_LOGFILE, DATE_FORMAT_WITH_SECONDS } from '../../../../Constants.js';
 import { blastOff } from '../../../../functions/anti-raid/blastOff.js';
 import { formatMemberTimestamps } from '../../../../functions/anti-raid/formatMemberTimestamps.js';
+import type { Case } from '../../../../functions/cases/createCase.js';
+import type { AntiRaidArgsUnion } from '../../../../functions/formatters/generateAntiRaidNukeReport.js';
```
Comment (iCrawl) [inline packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:22]:
```suggestion
import type { AntiRaidNukeArgsUnion } from '../../../../functions/formatters/generateAntiRaidNukeReport.js';
```

File: packages/yuudachi/src/functions/logging/upsertGeneralLog.ts [hunk 19]
```diff
@@ -1,21 +1,62 @@
-import type { Guild, User } from 'discord.js';
+import dayjs from 'dayjs';
+import { ButtonStyle, ComponentType, Guild, User } from 'discord.js';
```
Comment (iCrawl) [inline packages/yuudachi/src/functions/logging/upsertGeneralLog.ts:2]:
```suggestion
import { ButtonStyle, ComponentType, type Guild, type User } from 'discord.js';
```

File: packages/yuudachi/src/functions/logging/upsertGeneralLog.ts [hunk 20]
```diff
@@ -1,21 +1,62 @@
...
 import i18next from 'i18next';
 import { generateAntiRaidNukeReportEmbed } from './generateAntiRaidNukeReportEmbed.js';
+import { DATE_FORMAT_LOGFILE } from '../../Constants.js';
+import { createMessageActionRow } from '../../util/messageActionRow.js';
 import type { AntiRaidNukeResult } from '../anti-raid/blastOff.js';
+import { FormatterArgs, generateAntiRaidNukeReport } from '../formatters/generateAntiRaidNukeReport.js';
```
Comment (iCrawl) [inline packages/yuudachi/src/functions/logging/upsertGeneralLog.ts:8]:
```suggestion
import { type FormatterArgs, generateAntiRaidNukeReport } from '../formatters/generateAntiRaidNukeReport.js';
```

**Review by iCrawl — DISMISSED** [review]

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts (hunk 3 above)
Comment (JPBM135) [reply packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:14]: Yeah sorry for the type imports 😅, I will look closely after them next time

File: packages/yuudachi/locales/en-US/translation.json (hunk 4 above)
Comment (JPBM135) [reply packages/yuudachi/locales/en-US/translation.json:462]: `{{- id}} |`
          ^
The removal of the space here is intentional and will be fine bc of the padding

File: packages/yuudachi/src/functions/formatters/generateAntiRaidNukeReport.ts (hunk 16 above)
Comment (JPBM135) [reply packages/yuudachi/src/functions/formatters/generateAntiRaidNukeReport.ts]: Fixed in 55eea47

File: packages/yuudachi/locales/en-US/translation.json (hunk 4 above)
Comment (JPBM135) [reply packages/yuudachi/locales/en-US/translation.json]: Fixed in 731a286

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts (hunk 2 above)
Comment (JPBM135) [reply packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts]: Fixed in 67ee213

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts (hunk 1 above)
Comment (JPBM135) [reply packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts]: Fixed in 67ee213

**Comment by JPBM135** [comment]
Gh bugged again, cannot find any unresolved reviews

**Pull request merged**